### PR TITLE
Do not require `format` for `VarFiles`

### LINKED ISSUE #27: Validation error when no 'format' specified
### What happened?
According to https://github.com/crossplane-contrib/provider-terraform/blob/master/examples/workspace-remote.yaml#L23-L34 the `format` should be omittable and default to HCL

It does not happen in reality
```sh
error: error validating "examples/terraform/eks/eks.yaml": error validating data: ValidationError(Workspace.spec.forProvider.varFiles[0]): missing required field "format" in io.crossplane.tf.v1alpha1.Workspace.spec.forProvider.varFiles; if you choose to ignore these errors, turn validation off with --validate=false
```

### How can we reproduce it?
Apply Workspace resource with VarFile without `format` explicitly specified
Example: https://github.com/crossplane-contrib/provider-terraform/blob/master/examples/workspace-remote.yaml#L22-L26

### PR DESCRIPTION
**Pull request by ytsarev**
* Make `format` optional with default to `HCL`
* Fixes #27

Signed-off-by: Yury Tsarev <yury.tsarev@absa.africa>

### Description of your changes

Fixes #27 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

end-to-end
[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/v1alpha1/workspace_types.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ type VarFile struct {
 
 	// Format of this vars file.
 	// +kubebuilder:default=HCL
+	// +optional
 	Format VarFileFormat `json:"format"`
```
Comment (negz) [inline apis/v1alpha1/workspace_types.go:59]: We should make this a pointer since it's now `+optional`, per Kube API conventions.

Comment (ytsarev) [reply apis/v1alpha1/workspace_types.go:59]: @negz are you sure it is valid for `string` not `struct` 
I checked https://book.kubebuilder.io/cronjob-tutorial/api-design.html, see ConcurrencyPolicy there which is very similar to `VarFileFormat`, no pointers

Comment (negz) [reply apis/v1alpha1/workspace_types.go:59]: Yes - this applies to strings too. Or more specifically "any type without a built-in nil value" per https://github.com/kubernetes/community/blob/3cb570/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required. See also https://github.com/kubernetes/community/blob/3cb570/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required which has at least one `*string` in the example.

Comment (ytsarev) [reply apis/v1alpha1/workspace_types.go]: @negz thanks, does it look better now?

Comment (negz) [reply apis/v1alpha1/workspace_types.go]: It does, thank you!

**Review by negz — APPROVED** [review]

**Pull request merged**